acme_test.go

```diff
@@ -155,6 +159,81 @@ func TestAuthorize(t *testing.T) {
 	}
 }
 
+func TestAcceptChallenge(t *testing.T) {
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.Method == "HEAD" {
+			w.Header().Set("replay-nonce", "test-nonce")
+			return
+		}
+		if r.Method != "POST" {
+			t.Errorf("r.Method = %q; want POST", r.Method)
+		}
+
+		// Decode request
```

Comment:
This may be a bit overkill, but I think that testing that the right request is being formed is important. If you agree, I'll extract this and retroactively add it to the other functions, in a separate PR.